playbooks/adjust-cinder-backend.yml

```
@@ -0,0 +1,239 @@
+    - fail: msg="all.yml must not contain cinder.common_volume_name attribute."
+      when: cinder.common_volume_name is defined
+      run_once: true
+
+    - fail: msg="Ceph is not enabled for this cluster."
+      when: not ceph.enabled|default('False')|bool
+      run_once: true
+
+    - fail: msg="Cinder is not enabled for this cluster."
+      when: not cinder.enabled|default('False')|bool
+      run_once: true
+
+    - name: dump cinder db
+      mysql_db:
+        name: cinder
+        state: dump
+        target: /backup/cinder-preupgrade.{{ date_time }}.sql
+      run_once: true
+      delegate_to: "{{ groups['db'][0] }}"
+      changed_when: false
+
+  handlers:
+  - name: stop cinder services
+    service:
+      name: "{{ item }}"
+      state: stopped
+      must_exist: false
+    with_items: ['cinder-api', 'cinder-scheduler', 'cinder-volume']
+    when: set_cinder_host or update_cinder_backend
+
+  - name: stop cinder backup service
+    service:
+      name: cinder-backup
+      state: stopped
+      must_exist: false
+    when:
+      - swift.enabled|default('False')|bool
+      - (set_cinder_host or update_cinder_backend)
+
+  - name: restart cinder services
+    service:
+      name: "{{ item[1] }}"
+      state: restarted
+      must_exist: false
+    run_once: True
+    delegate_to: "{{ item[0] }}"
+    when: set_cinder_host or update_cinder_backend
+    with_nested:
+      - "{{ play_hosts }}"
+      - ['cinder-api', 'cinder-scheduler', 'cinder-volume']
+
+  - name: restart cinder backup service
+    service:
+      name: cinder-backup
+      state: restarted
+      must_exist: false
+    run_once: True
+    delegate_to: "{{ item }}"
+    when:
+      - swift.enabled|default('False')|bool
+      - (set_cinder_host or update_cinder_backend)
+    with_items:
+      - "{{ play_hosts }}"
+
+  tasks:
+  - name: gather facts for adjusting cinder service host
+    shell: grep '^host \?= \?ceph' /etc/cinder/cinder.conf
+    ignore_errors: true
```

We prefer `failed_when: false` instead of `ignore_errors`. It is easier to read the output as it will not color the output red if there was some (ignored) error.